src/main/kube-auth-proxy/kube-auth-proxy-cert.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { access, mkdir, writeFile } from "fs/promises";
+import path from "path";
+import * as selfsigned from "selfsigned";
+
+export function getKubeAuthProxyCertificate(): selfsigned.SelfSignedCert {
+  const opts = [
+    { name: "commonName", value: "Lens Certificate Authority" },
+    { name: "organizationName", value: "Lens" },
+  ];
+
+  return selfsigned.generate(opts, {
+    keySize: 2048,
+    algorithm: "sha256",
+    days: 365,
+    extensions: [
+      { name: "basicConstraints", cA: true },
+      { name: "subjectAltName", altNames: [
+        { type: 2, value: "localhost" },
+        { type: 7, ip: "127.0.0.1" },
+      ] },
+    ],
+  });
+}
+
+export function getKubeAuthProxyCertificatePath(baseDir: string) {
```

Comment:
Thought: a convention is to use `function` to advertise that `this` has special significance. Here it does not, and lambda could be used to advertise that instead.

Reply:
I would disagree, this codebase only uses `class`-es for `this` significant code and lambdas can sometimes make the code harder to read.